Lib/test/test_tempfile.py

```diff
@@ -39,27 +40,46 @@
 class TestLowLevelInternals(unittest.TestCase):
     def test_infer_return_type_singles(self):
         self.assertIs(str, tempfile._infer_return_type(''))
+        self.assertIs(str, tempfile._infer_return_type(pathlib.Path('')))
```

Comment:
I couldn't find other place test path-like with bytes. Add some test here, and at low-level internals test